File: replace_alts.js
const fs = require('fs');
const path = require('path');

const dirPath = '.';
const patternAlt = /(<img[^>]*?alt=)(['"])(.*?)\2/gi;
const patternImg = /<img[^>]+>/gi;

let count = 0;

function walk(dir) {
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('node_modules') && !file.includes('.git')) {
                walk(file);
            }
        } else if (file.endsWith('.html')) {
            let content = fs.readFileSync(file, 'utf8');
            let newContent = content.replace(patternImg, (imgTag) => {
                // If it already has an alt tag, replace its value
                if (/alt=['"]/i.test(imgTag)) {
                    return imgTag.replace(/(alt=)(['"])(.*?)\2/i, `$1$2Household Packers$2`);
                } else {
                    // If it doesn't have an alt tag, add it before the closing bracket
                    if (imgTag.endsWith('/>')) {
                        return imgTag.slice(0, -2) + ' alt="Household Packers" />';
                    } else {
                        return imgTag.slice(0, -1) + ' alt="Household Packers">';
                    }
                }
            });
            if (content !== newContent) {
                fs.writeFileSync(file, newContent, 'utf8');
                count++;
            }
        }
    });
}

walk(dirPath);
console.log(`Modified ${count} files`);

File: replace_alts.py
import os
import re

dir_path = '.'
pattern = re.compile(r'(<img[^>]*?alt=)([\'\"])(.*?)\2', re.IGNORECASE | re.DOTALL)

count = 0
for root, dirs, files in os.walk(dir_path):
    # skip node_modules
    if 'node_modules' in root:
        continue
    for file in files:
        if file.endswith('.html'):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
            except UnicodeDecodeError:
                continue
            
            # The regex will capture:
            # 1: `<img ... alt=`
            # 2: quote char `"` or `'`
            # 3: existing alt text
            
            def replacer(match):
                prefix = match.group(1)
                quote = match.group(2)
                return f'{prefix}{quote}Household Packers{quote}'
                
            new_content = pattern.sub(replacer, content)
            
            if new_content != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                count += 1

print(f'Modified {count} files')
